File: pset2/initials/initials.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>

char* getPersonsName(void);
void extractInitials(char*);

// more comfortable
int main(void)
{
    extractInitials(getPersonsName());
    return 0;
}

// Get name from user
char* getPersonsName(void)
{
	char* s = get_string();
	
    if (s != NULL) {
        return s;
    }
	else {
	    return "";
	}
}

// Extract initials from giving string
void extractInitials(char* p)
{
    for (int i = 0, n = strlen(p); i < n; i++) {
        if (isalpha(p[i])) {
            if (i == 0 || !isalpha(p[i - 1])) {
                printf("%c", toupper(p[i]));
            }
        }
    }
    printf("\n");
}

File: pset6/initials/initials.go
package main

import (
	"bufio"
	"bytes"
	"fmt"
	"os"
	"strings"
)

// get name from user
func getPersonsName() (string, error) {

	// constructs a new Reader
	reader := bufio.NewReader(os.Stdin)

	// get entire string of input
	line, err := reader.ReadString('\n')
	if err != nil {
		return "", err
	}

	return line, nil
}

// extract initials from giving string
func extractInitials(line string, err error) ([]byte, error) {

	// if incorrect
	if err != nil {
		return nil, err
	}

	// splits the string
	field := strings.Fields(line)

	var result []byte
	for _, v := range field {
		// сonverts a string transform character to upper case
		result = append(result, bytes.ToUpper([]byte(v))[0])
	}

	return result, nil
}

// more comfortable
func main() {

	if res, err := extractInitials(getPersonsName()); err != nil {

		fmt.Println(err)
	} else {
		// result with new line
		fmt.Printf("%s\n", res)
	}
}
